### DISCUSSION TITLE
Problem with image and note

### DISCUSSION THREAD

**Discussion by Magic-King** [TriliumNext/Trilium#1425] [Q&A]
I have a question that why sometimes when I clip the image to one note of trilium, it will sometimes be saved as the child note,but sometimes it will not. I wanna know how I copy the image into the note  so that the image wouldn't be the note's childnote?

**Comment by zadam** [comment]
Hi, images are always stored as child of some note.

What you may be observing is this setting:
![grafik](https://user-images.githubusercontent.com/617641/99254409-9e94b400-2812-11eb-862d-1908bc747194.png)

If the note links to the image, then it's by default hidden in the tree.

**Comment by Magic-King** [comment]
![image](https://user-images.githubusercontent.com/44387077/99271970-353c9180-2862-11eb-813b-004dbc7b62c6.png)
emmmmmmmmm
As you can see， I have checked this option , but some image still exist in the child note.
By the way , do the images in the note if I delete the image in the child note of the note?

**Reply by zadam** [reply]
They will show up unless parent note links to them. And they will still exists there anyway, it's just hiding them.

> By the way , do the images in the note if I delete the image in the child note of the note?

There is only one image (in the child note) and parent note only links to it. If you delete the child image then it will not be available in the parent either.

**Reply by Magic-King** [reply]
> They will show up unless parent note links to them. And they will still exists there anyway, it's just hiding them.
> 
> > By the way , do the images in the note if I delete the image in the child note of the note?
> 
> There is only one image (in the child note) and parent note only links to it. If you delete the child image then it will not be available in the parent either.

Thank you for your patience. I have known how it runs.

**Comment by Magic-King** [comment]
Sorry about the problems coming again.
I found sometimes the image of note is not linked to the note. But it wouldn't happen when I just finish the note.
I have seen the log but  it's so normal.I am not sure if this is a bug.
On the other hand, I wanna know how to relink the image and the note which display it?

P.S. And now I found it would likely happen when I add the label `#readOnly`  to the note.

**Reply by zadam** [reply]
> I found sometimes the image of note is not linked to the note.

What do you mean by that?

**Reply by Magic-King** [reply]
> > I found sometimes the image of note is not linked to the note.
> 
> What do you mean by that?

Actually, it's mean that the hiden image of the note will be displayed as the child of the note  when I am going to modify the label. After the display of image,I found there is no link between the note and the image of the note.But the image in the note still points to the image in the child note.( with `hides image included in the note` checked)

**Reply by Magic-King** [reply]
> > I found sometimes the image of note is not linked to the note.
> 
> What do you mean by that?

When I add label ReadOnly for the note, the link between image of this note and the note disappears in the linkmap of the note.How should I solve this problem?
